### PR TITLE
chore: update static legend and remove deprecated references to legend

### PR DESCRIPTION
**Pull request by TCL735**
In support of https://github.com/influxdata/ui/issues/806

- Changes allow saving a hidden static legend without nulling
- Remove deprecated "Legend"

### CODE REVIEW

File: src/common/schemas/GaugeViewProperties.yml [hunk 1]
```diff
@@ -42,9 +42,5 @@
       type: string
     tickSuffix:
       type: string
-    staticLegend:
-      $ref: "./StaticLegend.yml"
```
Comment (glinton) [inline src/common/schemas/GaugeViewProperties.yml:46]: assuming this removal of static legend was intentional

**Review by glinton — COMMENTED** [review]
Just need to remove the actual `src/common/schemas/Legend.yml` file (assuming nothing else is referencing it).

**Review by glinton — APPROVED** [review]
disregard the previous comment, i see the file is removed :man_facepalming:

File: src/common/schemas/GaugeViewProperties.yml (hunk 1 above)
Comment (TCL735) [reply src/common/schemas/GaugeViewProperties.yml:46]: Yes this is intentional. Gauge will not have a static legend.

**Pull request merged**